app-search-node.d.ts

```diff
@@ -0,0 +1,812 @@
+     * https://www.elastic.co/guide/en/app-search/current/schema.html#schema-read
+     *
+     * @param engineName unique Engine name
+     * 
+     * @example
+     *
+     * const engineName = "favorite-videos";
+     *
+     * try {
+     * const schema = await client.getSchema(engineName);
+     *   console.log(schema);
+     * } catch (e) {
+     *   console.error(e);
+     * }
+     */
+    getSchema(engineName: string): Promise<Record<string, any>>;
+
+    /**
+     * Update schema for the current engine
+     *
+     * https://www.elastic.co/guide/en/app-search/current/schema.html#schema-patch
+     *
+     * @param engineName unique Engine name
+     * @param schema body of schema object
+     * 
+     * @example
+     *
+     * const engineName = "favorite-videos";
+     * const schema = {
+     *   views: "number",
+     *   created_at: "date",
+     * };
+     *
+     * try {
+     *   const updatedSchema = await client.updateSchema(engineName, schema);
+     *   console.log(updatedSchema);
+     * } catch (e) {
+     *   console.error(e);
+     * }
+     */
+    updateSchema(
+      engineName: string,
+      schema: Record<string, any>
+    ): Promise<Record<string, any>>;
+
+    /**
+     * Creates a jwt search key that can be used for authentication to enforce a set of required search options.
+     * 
+     * More details can be found [here](https://www.elastic.co/guide/en/app-search/current/authentication.html#authentication-signed).
+     *
+     * @param apiKey A public search key for your App Search Engine
+     * @param apiKeyName This name must match the name of the key above from your App Search dashboard
+     * @param options see the [App Search API](https://www.elastic.co/guide/en/app-search/current/authentication.html#authentication-signed) for supported search options
+     * @param signOptions see [node-jsonwebtoken](https://github.com/auth0/node-jsonwebtoken#usage) for supported sign options
+     * 
+     * @example
+     * 
+     * const publicSearchKey = 'search-xxxxxxxxxxxxxxxxxxxxxxxx'
+     * // This name must match the name of the key above from your App Search dashboard
+     * const publicSearchKeyName = 'search-key'
+     * const enforcedOptions = {
+     *   result_fields: { title: { raw: {} } },
+     *   filters: { world_heritage_site: 'true' }
+     * }
+
+     * const signOptions = {
+     *   expiresIn: '5 minutes'
+     * }
+
+     * const signedSearchKey = AppSearchClient.createSignedSearchKey(
+     *   publicSearchKey,
+     *   publicSearchKeyName,
+     *   enforcedOptions,
+     *   signOptions
+     * )
+
+     * const baseUrlFn = () => 'http://localhost:3002/api/as/v1/'
+     * const client = new AppSearchClient(undefined, signedSearchKey, baseUrlFn)
+
+     * client.search('sample-engine', 'everglade')
+     */
+    static createSignedSearchKey(
+      apiKey: string,
+      apiKeyName: string,
+      options?: Record<string, any>,
+      signOptions?: Record<string, any>
+    ): void;
```

Comment:
This method returns `jwt.sign`, which returns a `string` as it is used in this lib. `void` is only returned when a callback is given.

Reply:
Thanks for pointing that out, I'll update it.